include/mouse.h

```diff
@@ -21,21 +22,102 @@
 
 #include "dosbox.h"
 
-void Mouse_ShowCursor(void);
-void Mouse_HideCursor(void);
+// ***************************************************************************
+// Notifications from external subsystems - all should go via these methods
+// ***************************************************************************
 
-bool Mouse_SetPS2State(bool use);
+void MOUSE_EventMoved(int32_t x_rel, int32_t y_rel, int32_t x_abs, int32_t y_abs, bool is_captured);
+void MOUSE_EventPressed(uint8_t idx);
+void MOUSE_EventReleased(uint8_t idx);
+void MOUSE_EventWheel(int32_t w_rel);
 
-void Mouse_ChangePS2Callback(uint16_t pseg, uint16_t pofs);
+void MOUSE_SetSensitivity(int32_t sensitivity_x, int32_t sensitivity_y);
+void MOUSE_NewScreenParams(uint16_t clip_x, uint16_t clip_y, uint16_t res_x, uint16_t res_y,
+                           bool fullscreen, int32_t x_abs, int32_t y_abs);
+void MOUSE_NotifyMovedFake(); // for VMware protocol support
 
+// ***************************************************************************
+// Common structures, please only update via functions above
+// ***************************************************************************
 
-void Mouse_CursorMoved(float xrel,float yrel,float x,float y,bool emulate);
-void Mouse_CursorSet(float x,float y);
-void Mouse_ButtonPressed(uint8_t button);
-void Mouse_ButtonReleased(uint8_t button);
+typedef struct MouseInfoConfig {
 
-void Mouse_AutoLock(bool enable);
-void Mouse_BeforeNewVideoMode();
-void Mouse_AfterNewVideoMode(bool setmode);
+    float  sensitivity_x;    // sensitivity, might depend on the GUI/GFX 
+    float  sensitivity_y;    // for scaling all relative mouse movements
+
+    MouseInfoConfig() :
+        sensitivity_x(0.3f),
+        sensitivity_y(0.3f)
+    {}
+
+} MouseInfoConfig;
```

Comment:
Can drop trailing type name as well.

Here's how it works:

``` c++
#include <stdio.h>

struct MouseInfoConfig {
    float sensitivity_x = 0.3f;    // sensitivity, might depend on the GUI/GFX 
    float sensitivity_y = 0.3f;    // for scaling all relative mouse movements
};

int main()
{
    MouseInfoConfig m = {};
    
    printf("default sensitivity_x = %f\n", m.sensitivity_x);
    printf("default sensitivity_y = %f\n", m.sensitivity_y);

    return 0;
}
```

``` text
default sensitivity_x = 0.300000
default sensitivity_y = 0.300000


...Program finished with exit code 0
Press ENTER to exit console.
```

https://onlinegdb.com/_lvNkHpc9

Reply:
I have turned this into the class. Unfortunately, not every compiler liked the struct with initializers :( Initialization in MOUSE_Init is out of question - this is too late, SDL subsystem starts up first. Besides, I don't want to rely on order of initialization, this is asking for trouble.

Turned this into a class with everything public, this should do the trick.